vendor/wp-now/src/wp-now.ts

```diff
@@ -471,6 +459,36 @@ export function getThemeTemplate( projectPath: string ) {
 async function mountInternalMuPlugins( php: PHP ) {
 	php.mkdir( PLAYGROUND_INTERNAL_MU_PLUGINS_FOLDER );
 
+	php.writeFile(
+		path.posix.join( PLAYGROUND_INTERNAL_MU_PLUGINS_FOLDER, '0-https-for-reverse-proxy.php' ),
+		`<?php
+		// See https://developer.wordpress.org/advanced-administration/security/https/#using-a-reverse-proxy
+		if( isset( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) && strpos( $_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') !== false ){
+			$_SERVER['HTTPS'] = 'on';
+		}
+		`
+	);
+
+	php.writeFile(
+		path.posix.join( PLAYGROUND_INTERNAL_MU_PLUGINS_FOLDER, '0-redirect-to-siteurl-constant.php' ),
+		`<?php
+		// See https://core.trac.wordpress.org/ticket/33821#comment:10
+		add_action( 'init', function() {
+			if ( ! defined( 'WP_SITEURL' ) ) {
+				return;
+			}
+
+			$current_host = $_SERVER['HTTP_HOST'] ?? '';
+
+			if ( preg_match( '/^localhost:\\d+$/', $current_host ) ) {
+				$requested_uri = $_SERVER['REQUEST_URI'] ?? '/';
+				wp_redirect( rtrim( WP_SITEURL, '/' ) . $requested_uri, 302 );
+				exit;
+			}
+		});
+		`
+	);
```

Comment:
On 2255b7f354d4652661625d5530d39b1c6e37d9f4 we had a function `geWpConfigConstants` that exposed wp-config.php constants to Studio to display the correct links in wp-admin and open site.

After discussing it with @wojtekn , we decided to simplify the logic and relay in WordPress default redirect. Due to this issue https://core.trac.wordpress.org/ticket/33821#comment:10 the redirect doesn't work out of the box and we need this mu plugin.

The wp-admin link redirects correctly to the WP_SITEURL/ngrok url, but it doesn't perform the autologin.

Reply:
@fredrikekelund correct. It seems it also doesn't handle non-ssl `http://localhost:8888/` redirect to `https://my-tunnel.com/` well. It would make sense to dedicate some time later to check what we could do rely on the core `redirect_canonical`.